test/e2e/12_fluxctl_config.bats

```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env bats
+
+function setup() {
+  load lib/env
+  load lib/install
+  load lib/poll
+
+  kubectl create namespace "$FLUX_NAMESPACE"
+  install_git_srv
+  tmp_config=$(mktemp -d)
+  defer rm -rf "$tmp_config"
+  cat > "${tmp_config}/flux.yaml" << EOF
+flux-config-version: v1
+EOF
+  install_flux_with_fluxctl '12_flux_config' '' --config-file="${tmp_config}/flux.yaml"
+}
+
+@test "'fluxctl install --config-file=...' smoke test" {
```

Comment:
I would add a couple of scenarios:

1. Setting an actual configuration value as opposed to a an empty config file and making sure it reaches Flux.
2. Checking the precedence of flags over configuration file entries by setting a configuration option in both places.

A good candidate for this is the flux URL which can be retrieved with `fluxctl sync` (alternatively could maybe just `curl` into `/v9/git-repo-config` opening the door to the branch etc ... )

Reply:
I would add a third one:

3. Test that the default value of a configuration option is set appropriately (with flags and configuration files handling default values doesn't seem straightforward)

Reply:
This hasn't been addressed